RxQuiz/ViewController/QuizViewController.swift

```diff
@@ -127,43 +126,57 @@ final class QuizViewController: UIViewController {
         }
         
         currentQuestionIndex.asDriver()
+            .filter { [unowned self] in $0 < self.questions.value.endIndex }
             .driveNext { [unowned self] count in
-                switch count {
-                case self.questions.value.endIndex:
-                    func restart() {
-                        self.questions.value.removeAll()
-                        self.currentQuestionIndex.value = 0
-                        self.correctAnswerCount.value = 0
-                        self.startDate = NSDate()
+                let question = self.questions.value[count]
+                self.currentQuestion.value = question
+                self.currentAnswer.value = question.answer
+            }
+            .addDisposableTo(disposeBag)
+        
+        currentQuestionIndex.asDriver()
+            .filter { [unowned self] in $0 == self.questions.value.endIndex }
+            .withLatestFrom(correctAnswerCount.asDriver()
+                .map { count -> String in
```

Comment:
I would be very grateful if you could explain the reason why return type is necessary when the switch statement is used. A compile error occurred when I omitted the return type but It was OK to omit it when I just wrote return String value (e.g. return "Test") without using the switch statement.